### PR TITLE
Monitor external dependency health

### PR DESCRIPTION
**Pull request by anthonyshull**
https://app.asana.com/0/555089885850811/1206498103588063/f

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,6 +138,29 @@ For details on environment configuration, including optional variables, see
...
+cd redis-7.2.4
+make
+./utils/create-cluster/create-cluster start
+./utils/create-cluster/create-cluster create -f
+```
+
+When you're done with it:
+
+```
+./utils/create-cluster/create-cluster start
```
Comment (thecristen) [inline README.md:159]: I'm guessing this one should be `stop`

File: integration/utils.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const axios = require("axios");
+
+const axiosOptions = {
+  headers: {
+    "User-Agent": "Node",
```
Comment (thecristen) [inline integration/utils.js:5]: praise: thank you for the custom header! 
nit: maybe a better custom name would be "X-Dotcom-Integration"

**Review by thecristen — CHANGES_REQUESTED** [review]
Just the OTP URL variable name is blocking, otherwise I feel like we're good to go here.

File: integration/e2e_tests/all-scenarios.spec.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  ? `https://${process.env.HOST}`
+  : "http://localhost:4001";
+
+files.forEach((file) => {
+  test.describe("All scenarios", (_) => {
+    const filePath = path.join(filesPath, file);
+    const { scenario } = require(filePath);
+
+    const name = fileToMetricName(file);
+
```
[the remark this answers is not in the document]
Comment (anthonyshull) [reply integration/e2e_tests/all-scenarios.spec.js:21]: We are using the scenarios for the monitor and load tests as well which is why we pass it into the scenario.

File: integration/utils.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    console.error({ baseURL: options.baseURL, error: e.message });
+
+    healthy = false;
+  }
+
+  return healthy;
+};
+
+const fileToMetricName = (file) =>
+  file.replace(/-/g, ".").replace(".js", "").toLowerCase();
```
[the remark this answers is not in the document]
Comment (anthonyshull) [reply integration/utils.js:30]: You can only use ASCII alphanumerics, periods, and underscores. We could convert the name to use underscores, but the metric category should be separated with a period. So, `dotcom.monitor.find_transit_near_me`. But, some checks only have one word so they would be `dotcom.monitor.healthcheck.cms`. So, I figured to just stick with periods.

File: integration/scenarios/visit-the-homepage.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const { expect } = require("@playwright/test");
+
+exports.scenario = async ({ page, baseURL }) => {
+  await page.goto(`${baseURL}/`);
+
+  await expect(
+    page.getByRole("heading", { name: "MBTA Homepage" }),
```
[the remark this answers is not in the document]
Comment (anthonyshull) [reply integration/scenarios/visit-the-homepage.js:7]: It's not "visible", but it is in the HTML. There is a different method to check if something is in the viewport. But, here, we just want to make sure the page is loaded.

https://playwright.dev/docs/actionability#visible

File: integration/utils.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+const axios = require("axios");
```
[the remark this answers is not in the document]
Comment (anthonyshull) [reply integration/utils.js:1]: Node 21 is the first release where fetch is stable. I can switch it out. I'm not sure what the benefit is at this point.

File: integration/utils.js (hunk 2 above)
Comment (anthonyshull) [reply integration/utils.js:5]: `X-...` is the convention for the name of a custom header, not the value. We're distinguishing between `Node`, `Artillery`, and `Playwright`. We can change `Node` to `Monitor` or something but the idea is to name the `User-Agent` with the agent which is just a node application.

File: integration/utils.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (anthonyshull) [reply integration/utils.js:1]: Also, the Fetch API looks gnarly. They don't have a simple mechanism to just timeout the request like axios does. Instead you have to use an abort controller with a setTimeout call.

**Review by thecristen — APPROVED** [review]
🚀

**Pull request merged**